plugins/kubernetes/app/models/concerns/kubernetes/deploy_yaml.rb

```diff
@@ -0,0 +1,61 @@
+module Kubernetes
+  module DeployYaml
+    def deployment_hash
+      @deployment_yaml ||=
+        if replication_controller_doc.present?
+          JSON.parse(replication_controller_doc).with_indifferent_access
```

Comment:
Yea, when redeploying an old release was the original use case, I just copied this code. Scaling it will probably need to regenerate the YAML rather then using what was there.